File: src/main/java/net/cazzar/corelib/util/ClientUtil.java
/*
 * The MIT License (MIT)
 *
 * Copyright (c) 2014 Cayde Dixon
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

/**
 *
 */
package net.cazzar.corelib.util;

import net.minecraft.client.Minecraft;
import net.minecraftforge.fml.common.FMLCommonHandler;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

/**
 * @author Cayde
 */
public class ClientUtil {
    /**
     * Get the minecraft instance
     *
     * @return the Minecraft instance
     */
    @SideOnly(Side.CLIENT)
    public static Minecraft mc() {
        return Minecraft.getMinecraft();
    }

    /**
     * Is this the client
     *
     * @return true if it is a client.
     */
    public static boolean isClient() {
        return FMLCommonHandler.instance().getSide().isClient();
    }

    /**
     * Get if it is single player or not
     *
     * @return true if single player
     */
    @SuppressWarnings("UnusedDeclaration")
    @SideOnly(Side.CLIENT)
    public static boolean isSinglePlayer() {
        return mc().isSingleplayer();
    }
}

File: src/main/scala/net/cazzar/corelib/client/util/ClientUtil.scala
package net.cazzar.corelib.client.util

import cpw.mods.fml.relauncher.{SideOnly, Side}
import net.minecraft.client.Minecraft
import cpw.mods.fml.common.FMLCommonHandler

object ClientUtil {
  /**
   * Get the minecraft instance
   *
   * @return the Minecraft instance
   */
  @SideOnly(Side.CLIENT) def mc: Minecraft = {
    Minecraft.getMinecraft
  }

  /**
   * Is this the client
   *
   * @return true if it is a client.
   */
  def isClient: Boolean = {
    FMLCommonHandler.instance.getSide.isClient
  }

  /**
   * Get if it is single player or not
   *
   * @return true if single player
   */
  @SuppressWarnings("UnusedDeclaration")
  @SideOnly(Side.CLIENT) def isSinglePlayer: Boolean = {
    mc.isSingleplayer
  }
}
